### ISSUE TITLE
ImportError: cannot import name 'python' from 'mediapipe.tasks.python' when building from official DockerFile

### ISSUE THREAD

**Issue by thomthomgo** [google/mediapipe#4657]
### OS Platform and Distribution

Ubuntu 20.04.6 LTS

### Compiler version

cmake version 3.16.3 / gcc and g++ v8.4.0

### Programming Language and version

Python 3.10

### Installed using virtualenv? pip? Conda?(if python)

pip

### MediaPipe version

latest (from master)

### Bazel version

6.1.1

### Android AAR (if android)

None

### OpenCV version (if running on desktop)

opencv-contrib-python==4.8.0.74 (installed from setup_opencv.sh)

### Describe the problem

```shell
Hi ! I'm trying to use the official Dockerfile but I'm eventually running into an issue : `ImportError: cannot import name 'python' from 'mediapipe.tasks.python' (/my_env/lib/python3.8/site-packages/mediapipe/tasks/python/__init__.py)`.

Here are the steps I went through:

* Cloned repo, built docker image and run it
* To follow [the instructions](https://developers.google.com/mediapipe/framework/getting_started/install#installing_on_debian_and_ubuntu), I had to first install sudo 
* Installed bazel 6.1.1 with bazelisk
* Ran setup_opencv.sh script to get openCV, but it complained about missing cmake, so I installed it and ran the script again
* I then set-up virtualenv with `apt install python3.8-venv`, created a virtual environment and started it, ran `pip install -r requirements.txt` and tried to execute `python3 setup.py install —link-opencv`
* It complained about missing protoc, so I installed it following [their official instructions](https://grpc.io/docs/protoc-installation/#install-pre-compiled-binaries-any-os), and got version 3.15.8
* I ran `python3 setup.py install —link-opencv` again and it completed
* Running `pip freeze` I get:
    
    absl-py==1.4.0
    attrs==23.1.0
    cffi==1.15.1
    contourpy==1.1.0
    cycler==0.11.0
    flatbuffers==23.5.26
    fonttools==4.41.1
    importlib-resources==6.0.0
    kiwisolver==1.4.4
    matplotlib==3.7.2
    mediapipe===dev
    numpy==1.24.4
    opencv-contrib-python==4.8.0.74
    packaging==23.1
    Pillow==10.0.0
    protobuf==3.20.3
    pycparser==2.21
    pyparsing==3.0.9
    python-dateutil==2.8.2
    six==1.16.0
    sounddevice==0.4.6
    zipp==3.16.2
    ```
* I then tried to import mediapipe within a script but get the following error:
    ```
    Traceback (most recent call last):
    File "<stdin>", line 1, in <module>
    File "/my_env/lib/python3.8/site-packages/mediapipe/__init__.py", line 30, in <module>
        import mediapipe.tasks.python as tasks
    ImportError: cannot import name 'python' from 'mediapipe.tasks.python' (/my_env/lib/python3.8/site-packages/mediapipe/tasks/python/__init__.py)
    ```
```


### Complete Logs

```shell
Will attach
```

**Comment by thomthomgo** [comment]
Please find the logs from `python3 setup.py install —link-opencv` [mp install log](https://github.com/google/mediapipe/files/12226814/mp.install.log)

**Comment by kuaashish** [comment]
@thomthomgo,

Could you **import tasks API** as described in the [Python Setup ](https://developers.google.com/mediapipe/solutions/setup_python) Or As snippet code mentioned below:


```
import mediapipe

from mediapipe.tasks import python as tasks
```

Also, Please let us know if this works for you. Thank you!!

**Comment by thomthomgo** [comment]
Thanks @kuaashish, unfortunately this new line fails too:

```
  File "/dica_env/lib/python3.8/site-packages/mediapipe/__init__.py", line 30, in <module>
    import mediapipe.tasks.python as tasks
ImportError: cannot import name 'python' from 'mediapipe.tasks.python' (/dica_env/lib/python3.8/site-packages/mediapipe/tasks/python/__init__.py)
```

**Comment by kuaashish** [comment]
@thomthomgo,

Could you please confirm that you are trying to import as snippet code mentioned in the above [comment](https://github.com/google/mediapipe/issues/4657#issuecomment-1661544005)? We are able to import tasks API with **Python Version 3.10.12**, **GCC 11.4.0**, **MP v0.10.3** using the same command as screenshot attached below:

**Command:**

```
import mediapipe

from mediapipe.tasks import python as tasks
```
**Screenshot:**

<img width="722" alt="Screenshot 2023-08-03 at 12 14 55 PM" src="https://github.com/google/mediapipe/assets/98159216/6e910415-d446-44d3-a7a8-596aa70daabd">



Please let us know further update. Thank you

**Comment by thomthomgo** [comment]
Here is the result of the commands below. Please note that I'm using the [official DockerFile](https://github.com/google/mediapipe/blob/master/Dockerfile) as a starting point.
<img width="1440" alt="image" src="https://github.com/google/mediapipe/assets/27781457/0b00fac3-743e-4a05-806a-4b4bd2357e2d">

**Comment by zihaomu** [comment]
the same issue at Mac X86, python 3.10.

**Comment by kuaashish** [comment]
@thomthomgo, @zihaomu

Thank you both for detailed explanation, This looks like a genuine bug in our package. We are highlight issue internally And let us know if any update becomes available.

**Comment by s1n1o1w** [comment]
here I found that when my python version is 3.7, then "from mediapipe.tasks.python import vision" will not successful, 
but when I create a virtural env that python version is 3.9, then it works, 
in fact, the offical note that 
"  Supported platforms and versions
       Building applications with MediaPipe Tasks requires the following development environment resources:
 
       Python
            Linux/Mac/Windows: Python 3.8 - 3.11
"
so, maybe **the problem is the version of python**

**Comment by thomthomgo** [comment]
Sure, here it is:

```

from mediapipe.python import *
import mediapipe.python.solutions as solutions 
import mediapipe.tasks.python as tasks


del framework
del gpu
del modules
del python
del mediapipe
del util
__version__ = 'dev'

from mediapipe.python import *
import mediapipe.python.solutions as solutions 
import mediapipe.tasks.python as tasks


del framework
del gpu
del modules
del python
del mediapipe
del util
__version__ = 'dev'

from mediapipe.python import *
import mediapipe.python.solutions as solutions 
import mediapipe.tasks.python as tasks


del framework
del gpu
del modules
del python
del mediapipe
del util
__version__ = 'dev'
```

**Comment by e8035669** [comment]
Yeah, as you can see

The section below
```python
from mediapipe.python import *
import mediapipe.python.solutions as solutions 
import mediapipe.tasks.python as tasks


del framework
del gpu
del modules
del python
del mediapipe
del util
__version__ = 'dev'
```
is repeated for 3 times.
These lines should appear only once.

You should recover `mediapipe/__init__.py` file in your project first. This file is dirty.
Then rerun `python setup.py install`. The package will be better.

**Comment by e8035669** [comment]
https://github.com/google/mediapipe/blob/e10bcd1bfd9baf84aafb5929dfe29e6749830b68/setup.py#L133-L140

These code is **appending** some contents to the `__init__.py` file.
So this file will become dirty after a build, we should recover this file before next trial

**Comment by kuaashish** [comment]
@thomthomgo,


I could able to import mediapipe successfully by creating the **virtual environment** inside the docker as screenshot attached below:

<img width="600" alt="Screenshot 2023-08-10 at 5 12 42 PM" src="https://github.com/google/mediapipe/assets/98159216/531d50a5-9900-43e3-920c-a45097f29c00">


Please activate a **Python virtual environment** by using the **below command** or Using the guide [here](https://github.com/google/mediapipe/blob/master/docs/getting_started/python.md#ready-to-use-python-solutions):

`python3 -m venv mp_env && source mp_env/bin/activate`

And Install MediaPipe **Python package** and start **Python interpreter** Using the below command:


```
(mp_env)$ pip install mediapipe
(mp_env)$ python3
```

Also, Please let us know if this works for you. Thank you

**Comment by zihaomu** [comment]
Hi @kuaashish, the `pip install mediapipe` only works for 0.9 version, but not for 0.10.3. And after compiling from source code, `import mediapipe` failed.

**Comment by kuaashish** [comment]
@zihaomu,

Have you created the Virtual environment and tried to import package as suggested above? Also, We recommend you to raise a new issue with complete details to reproduce the issue from our end. Thank you!

**Comment by thomthomgo** [comment]
> Yeah, as you can see
> 
> The section below
> 
> ```python
> from mediapipe.python import *
> import mediapipe.python.solutions as solutions 
> import mediapipe.tasks.python as tasks
> 
> 
> del framework
> del gpu
> del modules
> del python
> del mediapipe
> del util
> __version__ = 'dev'
> ```
> 
> is repeated for 3 times. These lines should appear only once.
> 
> You should recover `mediapipe/__init__.py` file in your project first. This file is dirty. Then rerun `python setup.py install`. The package will be better.

This fixed my issue, thanks @e8035669 !

**Comment by matanox** [comment]
This file is code generated by setup.py. It tends to be generated with this block repeated in it, at least when you `pip install .` in now-older mediapipe code revisions, when using modern versions of pip.
